### ISSUE TITLE
.editorconfig fails to suppress analyzer warnings

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#42762]
**Version Used**: .NET Core SDK 3.1.201, VS 16.6.0 Preview 3 (29923.206.master)

**Steps to Reproduce**:

```
git clone https://github.com/AArnott/Nerdbank.Streams.git
cd Nerdbank.Streams
git checkout f5c2933e0b66b2a2ad7415c5d637ba5149d5370b
dotnet build src\Nerdbank.Streams
```

**Expected Behavior**:

No warnings for files under the `ext` directory.

**Actual Behavior**:

Many dozens of warnings under the `ext\MessagePack\src\MessagePack.UnityClient\Assetes\Scripts\MessagePack` direcotry.

The `ext` folder is where submodules are mounted, and while these add compilation inputs to the C# compiler, we can't hold them to the same style bar that our the parent repo is held to. For this reason, I have an `ext\.editorconfig` file that turns off *all* analyzers. But although I can see all the relevant .editorconfig files are passed to the compiler, the analyzers run and produce many warnings anyway.

![image](https://user-images.githubusercontent.com/3548/77551331-d8760c00-6e77-11ea-8c4d-93dcbf7333bc.png)

**Comment by AArnott** [comment]
Note that the VS in-proc compiler seems to do it right -- it doesn't produce these warnings in the error list except right after a full project build, and then when I navigate to those warnings in the editor, they disappear. So it seems it's just a bug in the command line build.

**Comment by AArnott** [comment]
Note if I move the content of `ext\.editorconfig` into the repo root `.editorconfig` file, the warnings go away.

**Comment by AArnott** [comment]
This repros with the very latest compiler as well:
```xml
    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.6.0-1.final" PrivateAssets="all" />
```

**Comment by jcouv** [comment]
Tagging @agocke

**Comment by ssa3512** [comment]
I think I am running into a similar issue - trying to turn on some of the analyzers for my project. I added the following to .`editorconfig` in the root of our repository. Building locally has the expected results (no errors/no CAxxxx warnings, except for CA1824 which may be a separate issue). Our Azure DevOps hosted CI build is giving lots of CA warnings despite the editorconfig

.editorconfig:
```editorconfig
# CA1827: Do not use Count/LongCount when Any can be used
dotnet_diagnostic.CA1827.severity = error

# CA1829: Use Length/Count property instead of Count() when available
dotnet_diagnostic.CA1829.severity = error

# Default severity for all analyzer diagnostics
dotnet_analyzer_diagnostic.severity = none
```

**Comment by AArnott** [comment]
I have [a persistent repro](https://onedrive.cloud.microsoft/:f:/a@qs3vp3zf/r/_layouts/15/onedrive.aspx?id=%2Fa%40qs3vp3zf%2FDocuments%2Fdumps%2FroslynIntellisenseWarningsBug42762&share=cgreaZ3d8a%5FsQYS2JKpkPJbWEgUCMSqQ5ZvdZFkfvIvlBz8fRQ) including devenv dump, binlogs of design-time builds, and repro steps.

This, despite the fact that CS1570 is suppressed in the .editorconfig file, which the design-time build binlog shows _is_ passed to the compiler.

<img width="1162" height="767" alt="Image" src="https://github.com/user-attachments/assets/0348911d-7e62-4d02-90ec-54bd90d9b66e" />

**Comment by jjonescz** [comment]
The original issue with Nerdbank.Streams (https://github.com/dotnet/roslyn/issues/42762#issue-587765340) doesn't repro with latest compiler. Using this issue to track the IDE-only issue reported in the comment https://github.com/dotnet/roslyn/issues/42762#issuecomment-4195714348.

**Comment by jjonescz** [comment]
The problem is that paths from MSBuild can come with wrong casing. But EditorConfig file matching is case sensitive ([by design](https://github.com/dotnet/roslyn/pull/82382#issuecomment-3894844276)). So e.g., `E:/SomeDir/.editorconfig` won't match for a file `E:/someDir/file.cs`, even though on case-insensitive file systems both are in the same directory. In the repro above, this mismatch is caused by a `ProjectReference` with the casing being non-canonical in the project path. Somehow (probably through project system), the IDE eventually gets `.editorconfig` with the non-canonical casing, and the C# files have canonical paths, causing a mismatch.

**Comment by jjonescz** [comment]
From offline discussion, this seems to be by design in the compiler. MSBuild should provide paths with correct casing. If the user provides paths with wrong casing in their project file, there would ideally be a warning. Tracked by https://github.com/dotnet/msbuild/issues/10968.
